plugins/@grouparoo/pardot/src/lib/client.ts

```diff
@@ -0,0 +1,215 @@
+import axios, { AxiosInstance } from "axios";
+import jsforce from "jsforce";
+
+interface PardotClientOptions {
+  businessUnitId: string;
+  loginUrl?: string;
+  pardotUrl?: string;
+}
+
+interface PardotCustomField {
+  id: number;
+  name: string;
+  fieldId: string;
+  type: string;
+  isRecordMultipleResponses: boolean;
+}
+
+class PardotClient {
+  private pardotUrl: string;
+  private businessUnitId: string;
+
+  conn: jsforce.Connection;
+  request?: AxiosInstance;
+
+  constructor(options?: PardotClientOptions) {
+    this.pardotUrl = options?.pardotUrl || "https://pi.pardot.com";
+    this.businessUnitId = options.businessUnitId;
```

Comment:
notably, you are optional in the above with `?` in the signature and url usage.
But then here's is required. Probably say it's required.

Reply:
are you referring to `businessUnitId`? It's always required here.

Reply:
Yeah, i was just saying in half this method you are acting like `options` might be null/undefined and in that spot not. So it will crash either way if it is null, so it is required, so it would be better to be more confident the other spots.
`options?.pardotUrl` -> `options.pardotUrl`